### DISCUSSION TITLE
Experience with HA core

### DISCUSSION THREAD

**Discussion by DaveDixon** [allangood/rtlamr2mqtt#240] [Q&A]
Has anyone used rtlamr2mqtt with HA core? I'm running HA on a linux server to interface with computation-intensive applications. Thanks!

**Comment by bokkusu** [comment]
~~I am trying to run rtlamr2mqtt with HA core but have not yet been successful. It appears that I do not have rtlarm2mqtt publishing stats to MQTT yet based on using MQTT Explorer to check. When I enable LISTEN_ONLY=yes and RTL_MSGTYPE="all" the container successfully finds and prints out broadcasts from my water meter to the container log so I believe I have the RTL-SDR hardware working correctly. And when I start and stop the container, the rtlamr/status topic updates between online and offline as appropriate so I believe I have MQTT set up correctly. But still there is no rtlamr/meter_water/state topic being published.~~

Edit: I have successfully configured Home Assistant Core to get data from rtlamr2mqtt. In actuality, HA Core is getting data from MQTT and rtlamr2mqtt is publishing its data to MQTT.

I am trying to do this on Unraid OS and I am running a mosquitto container to be the MQTT broker.

I have installed rtlamr2mqtt as a docker container manually. I have created rtlamr2mqtt.yaml from the sample included in the readme and configured the MQTT username and password credential and updated the meter ID to my meter's ID (printed on the unit or can be found using LISTEN_ONLY and RTL_MSGTYPE params). I also updated the meter type, this was what unblocked me. Use the meter type reported when using LISTEN_ONLY and RTL_MSGTYPE="all".

I have configured Home Assistant manually by editing its configuration.yaml as suggested in the readme to add a utility_meter integration. I attempted to set up a sensor as shown in the readme but Home Assistant called that an invalid configuration. It appears that the readme suggests an old way of specifying MQTT sensors and it has now changed.

The parts added to Home Assistant's confugration.yaml look like this:

```utility_meter:
  hourly_water:
    source: sensor.meter_water
    cycle: hourly
  daily_water:
    source: sensor.meter_water
    cycle: daily
  monthly_water:
    source: sensor.meter_water
    cycle: monthly


mqtt:
  sensor:
    - name: "meter_water"
      state_topic: "rtlamr/1248283751/state"```
